### PR TITLE
Add audit events for static host users

### PR DESCRIPTION
**Pull request by rosstimothy**
Emits audit events in response to users modifying or creating StaticHostUser resources.

Updates #42712

### CODE REVIEW

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -5426,6 +5426,7 @@ func NewGRPCServer(cfg GRPCServerConfig) (*GRPCServer, error) {
 
 	staticHostUserServer, err := userprovisioningv1.NewService(userprovisioningv1.ServiceConfig{
```
Comment (atburke) [inline lib/auth/grpcserver.go:5427]: Unrelated to your changes, but would you mind fixing this import?
```suggestion
	staticHostUserServer, err := userprovisioningv2.NewService(userprovisioningv1.ServiceConfig{
```

**Review by atburke — APPROVED** [review]

File: web/packages/teleport/src/Audit/EventList/EventTypeCell.tsx [hunk 2]
```diff
@@ -288,6 +288,9 @@ const EventIconMap: Record<EventCode, any> = {
   [eventCodes.INTEGRATION_CREATE]: Icons.Info,
   [eventCodes.INTEGRATION_UPDATE]: Icons.Info,
   [eventCodes.INTEGRATION_DELETE]: Icons.Info,
+  [eventCodes.STATIC_HOST_USER_CREATE]: Icons.Info,
```
Comment (zmb3) [inline web/packages/teleport/src/Audit/EventList/EventTypeCell.tsx:291]: Should we use a user icon?

Comment (kimlisa) [reply web/packages/teleport/src/Audit/EventList/EventTypeCell.tsx:291]: lets do it (i'm not sure why we didn't in the first place 🤔 )

**Review by kimlisa — APPROVED** [review]

**Review by kimlisa — COMMENTED** [review]
can you add the events to our storybook? 
you just need to define it here, and it should show up on [story](https://localhost:9002/?path=/story/teleport-audit--all-possible-events)

https://github.com/gravitational/teleport/blob/bc4215f29ecefe86146596259ea3192248efd74b/web/packages/teleport/src/Audit/fixtures/index.ts#L22

**Pull request merged**